src/main/java/com/google/api/codegen/discogapic/DiscoGapicProviderFactory.java

```diff
@@ -69,17 +70,18 @@
                 .setDiscoApiModel(model)
                 .setProductConfig(productConfig)
                 .setSnippetSetRunner(new CommonSnippetSetRunner(new JavaRenderingUtil()))
-                .setDocumentToViewTransformers(transformers)
+                .setModelToViewTransformers(transformers)
                 .build();
 
         providers.add(provider);
 
         CodeGenerator metadataProvider =
-            ViewModelDiscoGapicProvider.newBuilder()
-                .setModel(model)
+            DiscoGapicProvider.newBuilder()
```

Comment:
I didn't notice the duplication between `DiscoGapicProvider` and `ViewModelDiscoGapicProvider` before.

I'm thinking we should also rename `ViewModelGapicProvider` to `GapicProvider` to be consistent. (The `GapicProvider` name was freed up when I renamed that interface to `CodeGenerator`)

Taking the naming even further, we could instead use `DiscoGapicGenerator` and `GapicGenerator`. (And then, `DiscoGapicGeneratorFactory` and `GapicGeneratorFactory`...)

I'll leave that last suggestion up to you (it would be cleaner to do it in a following PR anyway I think)

Reply:
I will do this in a follow-up PR